File: Habitica/src/main/java/com/habitrpg/android/habitica/models/responses/HabitResponse.java
package com.habitrpg.android.habitica.models.responses;

import com.habitrpg.android.habitica.models.Notification;

import java.util.List;

/**
 * Created by krh12 on 11/23/2016.
 */

public class HabitResponse<T> {

    public T data;
    public List<Notification> notifications;
    private Boolean success;
    public String message;

    /**
     * @return The success
     */
    public Boolean getSuccess() {
        return success;
    }

    /**
     * @param success The success
     */
    public void setSuccess(Boolean success) {
        this.success = success;
    }

    /**
     * @return The data
     */
    public T getData() {
        return data;
    }
}

File: Habitica/src/main/java/com/habitrpg/android/habitica/models/responses/HabitResponse.kt
package com.habitrpg.android.habitica.models.responses

import com.habitrpg.android.habitica.models.Notification

class HabitResponse<T> {
    var data: T? = null
    var notifications: List<Notification>? = null
    var success: Boolean? = null
    var message: String? = null
}
